libraries/resource_firewall.rb

```diff
@@ -0,0 +1,19 @@
+class Chef
+  class Resource::Firewall < Resource
+    include Poise(container: true)
+
+    actions(:enable, :disable)
+    attribute(:log_level, :kind_of => Symbol, :equal_to => [:low, :medium, :high, :full], :default => :low)
+
+    def initialize(name, run_context=nil)
+      super(name, run_context)
+      [:ubuntu, :debian].each do |platform|
+        Chef::Platform.set(
```

Comment:
Should really be using the new Chef 12 provider mapping now. Or override `#provider_for_action`.